pkg/operator/controller/ingress/controller.go

```diff
@@ -1131,6 +1133,18 @@ func (r *reconciler) ensureIngressController(ci *operatorv1.IngressController, d
 
 	SetIngressControllerNLBMetric(ci)
 
+	// If the lbService exists, then update Infra CR's PlatformStatus with the Ingress LB IPs.
+	if haveLB {
+		if updated, err := updateInfraIngressLoadBalancerIPs(lbService, infraConfig); err != nil {
+			errs = append(errs, fmt.Errorf("failed to update Infrastructure PlatformStatus: %w", err))
+		} else if updated {
+			if err := r.client.Status().Update(context.TODO(), infraConfig); err != nil {
```

Comment:
I think this code should only update the infra config if this is the service for the "default" IngressController, right?  That is, you need something like `if haveLB && ci.Name == manifests.DefaultIngressControllerName`.  Otherwise, if you have a custom IngressController that uses `spec.endpointPublishingStrategy.type: LoadBalancerService`, the operator may keep updating the infra config status with the default IngressController's ingress IP addresses, then the custom IngressController's ingress IP addresses, then the default's, then the custom's, and so on.  (Sorry I didn't think of this during earlier reviews.)

Reply:
Done. 
Will the custom Ingress Controller's LB service use the cloud's default DNS then?

Reply:
Yeah, the cluster-admin will need to configure DNS records for the custom IngressController manually.  I think that's fine—the cluster-admin is expected to configure DNS records anyway, post-install, for the default IngressController, right?  And cluster-ingress-operator only needs to update the infra config so that the platform can provide DNS name resolution for internal clients connecting to platform routes (console, oauth) so that installation can complete?

Reply:
> And cluster-ingress-operator only needs to update the infra config so that the platform can provide DNS name resolution for internal clients connecting to platform routes (console, oauth) so that installation can complete?

Yes, that is correct.

> Yeah, the cluster-admin will need to configure DNS records for the custom IngressController manually.
That is fine. I just want to make sure we are not automatically configuring the cloud default DNS for the custom Ingress Controller case.